dagshub/data_engine/model/datasource.py

```diff
@@ -1064,3 +1195,119 @@ def wrap_bytes(val: bytes) -> str:
 
     def get_metadata_entries(self):
         return self._metadata_entries
+
+
+@dataclass
+class DatasourceQuery(DataClassJsonMixin):
+    as_of: Optional[int] = field(default=None, metadata=config(exclude=exclude_if_none, letter_case=LetterCase.CAMEL))
```

Comment:
Why is it an optional int if the args we accept for as_of are only datetime and float?
Shouldn't that mean we should accept ints as well?

Reply:
The int as_of is the one that's accepted by GraphQL, the datetime/float is the user facing one.